docs/source/error-codes/self-service/index.rst

```diff
@@ -7,6 +7,200 @@ Self-Service Error Codes (Experimental)
++------------------------------------------------+------------+--------------------+
+|DeadlineExceededRequestStateUnknown             |3           |DEADLINE_EXCEEDED   |
++------------------------------------------------+------------+--------------------+
+|SystemInternalAssumptionViolated                |4           |INTERNAL            |
++------------------------------------------------+------------+--------------------+
+|MaliciousOrFaultyBehaviour                      |5           |UNKNOWN             |
++------------------------------------------------+------------+--------------------+
+|AuthInterceptorInvalidAuthenticationCredentials |6           |UNAUTHENTICATED     |
++------------------------------------------------+------------+--------------------+
+|InsufficientPermission                          |7           |PERMISSION_DENIED   |
++------------------------------------------------+------------+--------------------+
+|InvalidIndependentOfSystemState                 |8           |INVALID_ARGUMENT    |
++------------------------------------------------+------------+--------------------+
+|InvalidGivenCurrentSystemStateOther             |9           |FAILED_PRECONDITION |
++------------------------------------------------+------------+--------------------+
+|InvalidGivenCurrentSystemStateResourceExists    |10          |ALREADY_EXISTS      |
++------------------------------------------------+------------+--------------------+
+|InvalidGivenCurrentSystemStateResourceMissing   |11          |NOT_FOUND           |
++------------------------------------------------+------------+--------------------+
+|InvalidGivenCurrentSystemStateSeekAfterEnd      |12          |OUT_OF_RANGE        |
++------------------------------------------------+------------+--------------------+
+|BackgroundProcessDegradationWarning             |13          |N/A                 |
++------------------------------------------------+------------+--------------------+
+
+
+Anatomy of an Error
+==========================
+
+
+Errors returned to users are represented as instances of standard StatusRuntimeException_.
+As such they contain a `gRPC status code`_, a description and additional machine readable information
+represented in the `rich gRPC error model`_.
+
+
+Error Description
+--------------------------
+
+We use the `standard gRPC description`_ that additionally adheres to our custom message format:
+
+.. code-block:: java
+
+    <ERROR_CODE_ID>(<CATEGORY_ID>,<TRUNCATED_CORRELATION_ID>):<HUMAN_READABLE_MESSAGE>
+
+The constituent parts are:
+
+  - ``<ERROR_CODE_ID>`` - a unique non empty string containing at most 63 characters:
+    upper-cased letters, underscores or digits.
+    Identifies corresponding error code id.
+
+  - ``<CATEGORY_ID>`` - a small integer identifying the corresponding error category.
+
+  - ``<TRUNCATED_CORRELATION_ID>`` - a string aimed at identifying originating request.
+    Contains at most 8 characters of the original correlation id.
```

Comment:
This is a prefix of the submission id

Reply:
Will add in a follow-up PR as now the CI is green. Wdyt?